src/sentry/search/events/builder.py

```diff
@@ -1645,12 +1660,133 @@ def __init__(
 
 
 class SessionsQueryBuilder(QueryBuilder):
+    # ToDo(ahmed): Rename this to AlertsSessionsQueryBuilder as it is exclusively used for crash rate alerts
     def resolve_params(self) -> List[WhereType]:
         conditions = super().resolve_params()
         conditions.append(Condition(self.column("org_id"), Op.EQ, self.organization_id))
         return conditions
 
 
+class SessionsV2QueryBuilder(QueryBuilder):
+    condition_fields = {"project", "project_id", "environment", "release"}
+
+    def __init__(
+        self,
+        *args: Any,
+        granularity: Optional[int] = None,
+        extra_condition_fields: Optional[Sequence[str]] = None,
+        **kwargs: Any,
+    ):
+        self._granularity = granularity
+        self._extra_condition_fields = extra_condition_fields or []
+        self._non_nullable_keys = self.condition_fields
+        # Fields that if present in the select, then need to be in the group by as well
+        self.required_select_groupby = copy.copy(self.condition_fields)
+        self.required_select_groupby.remove("project")
+        super().__init__(*args, **kwargs)
+
+    def resolve_params(self) -> List[WhereType]:
+        conditions = super().resolve_params()
+        conditions.append(Condition(self.column("org_id"), Op.EQ, self.organization_id))
+        return conditions
+
+    def resolve_groupby(self, groupby_columns: Optional[List[str]] = None) -> List[SelectType]:
```

Comment:
I know we discussed it in a meeting, but could you comment here explaining why the groupby for sessions needs to be handled differently?

Reply:
Yep